plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageSettings.java

```diff
@@ -43,20 +45,43 @@
         Setting.affixKeySetting(Storage.PREFIX, "key", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
     private static final Setting<Boolean> DEFAULT_SETTING =
         Setting.affixKeySetting(Storage.PREFIX, "default", (key) -> Setting.boolSetting(key, false, Setting.Property.NodeScope));
+    /** The host name of a proxy to connect to s3 through. */
+    private static final Setting<String> PROXY_HOST_SETTING =
+        Setting.affixKeySetting(Storage.PREFIX, "proxy.host", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
+    /** The port of a proxy to connect to azure through. */
+    private static final Setting<Integer> PROXY_PORT_SETTING = Setting.affixKeySetting(Storage.PREFIX, "proxy.port",
+        key -> Setting.intSetting(key, 80, 0, 1<<16, Setting.Property.NodeScope));
+    /** The type of the proxy to connect to azure through. Can be direct (no proxy, default), http or socks */
+    private static final Setting<Proxy.Type> PROXY_TYPE_SETTING = Setting.affixKeySetting(Storage.PREFIX, "proxy.type",
+            key -> new Setting<>(key, "direct", s -> Proxy.Type.valueOf(s.toUpperCase(Locale.ROOT)), Setting.Property.NodeScope));
 
 
     private final String name;
     private final String account;
     private final String key;
     private final TimeValue timeout;
     private final boolean activeByDefault;
-
-    public AzureStorageSettings(String name, String account, String key, TimeValue timeout, boolean activeByDefault) {
+    private final String proxyHost;
+    private final Integer proxyPort;
+    private final Proxy.Type proxyType;
+    private final Proxy proxy;
+
+    @SuppressForbidden(reason = "we know we pass a IP address or a host name")
```

Comment:
Why is this suppression here? The "reason" doesn't really explain anything. I assume it is because of call to construction of `InetSocketAddress` below. But this is not a valid reason to suppress; the message with forbidden apis clearly states you should look up the address of the host yourself with InetAddress, and then use that ctor for `InetSocketAddress`.